app/scripts/modules/core/src/presentation/modal/TabBoundary.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { useEventListener } from 'core';
+
+export const TabBoundary = ({ children }: { children: React.ReactNode }) => {
+  const dialogRef = React.createRef<HTMLDivElement>();
+
+  const keydownCallback = (event: KeyboardEvent) => {
+    const { keyCode, shiftKey, target } = event;
+    if (keyCode === 9) {
+      let targetOverride = null;
+      // tabbable selector via https://kennethbi.com/posts/5
+      const tabbableElements: HTMLElement[] = Array.from(
+        dialogRef.current.querySelectorAll(`
+          input:not([tabindex^="-"]):not([disabled]),
+          select:not([tabindex^="-"]):not([disabled]),
+          textarea:not([tabindex^="-"]):not([disabled]),
+          button:not([tabindex^="-"]):not([disabled]),
+          a[href]:not([tabindex^="-"]):not([disabled]),
+          [tabindex]:not([tabindex^="-"]):not([disabled])
+      `) as NodeListOf<HTMLElement>,
```

Comment:
ommgggg I did not know about `NodeListOf`, life changing for yucky DOM stuff. I've always used the parameter-less `NodeList` which is awful.

Reply:
Yeah, it's nice! I just stumbled into it here and it saved me a lot of Typescript awkwardness.